src/js/themes/base.d.ts

```diff
@@ -232,6 +232,7 @@ interface ButtonType {
       };
     };
   };
+  style?: Partial<CSSStyleDeclaration>;
```

Comment:
This feels like heading in a new direction for something that feels like we've used `extend` for in other places. What the rationale here?

Reply:
We still need `extend` when styles are driven from theme props, but if the theme uses component props it will also need to accommodate any `styles` of the component as well.